### PR TITLE
Hit the API

### PR DESCRIPTION
**Pull request by rpire**
### Changes made:

- Added functionality with new modules `get_board.js`, `gen_board.js`, and `add_new.js`.
- Built new async functions to get, generate, and add a new score to the leaderboard using the new modules.
- Added event listeners to `index.html`.
- Added functionality to buttons.

### CODE REVIEW

File: src/add_new.js [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
+  // scoreContainer.classList.add('item-score');
+  // scoreContainer.innerHTML = score;
 
-  leaderboardItem.appendChild(nameContainer);
-  leaderboardItem.appendChild(scoreContainer);
+  // leaderboardItem.appendChild(nameContainer);
+  // leaderboardItem.appendChild(scoreContainer);
 
-  leaderboard.appendChild(leaderboardItem);
+  // leaderboard.appendChild(leaderboardItem);
```
Comment (Emmyn5600) [inline src/add_new.js]: #### Optional suggestion

- [x] Kindly, remove the commented lines in your code to look more clean and professional

**Review by Emmyn5600 — APPROVED** [review]
Hi, 

Good job so far! 👏🏼 @rpire.

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- I can send and receive data to/from API.:heavy_check_mark:
- POST and GET method has been implemted to POST and GET data,:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the project requirements.
 
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/add_new.js (hunk 1 above)
Comment (rpire) [reply src/add_new.js]: Thank you! I totally forgot about the commented lines. I will make sure to remove them before merging.

**Comment by rpire** [comment]
Hello @Emmyn5600 👋

### Thank you! 🙌

Thank you for reviewing my work. I will make sure to implement the suggested changes before merging.

Cheers and Happy coding! 🚀

> Please note that there are in fact **three** branches, `main`, `dev`, and `hit-the-api`. And that the last one of these branches is merging the changes into `dev`. If there's any other thing that keeps me from following the GitFlow, please let me know.

File: src/add_new.js (hunk 1 above)
Comment (rpire) [reply src/add_new.js]: Changes completed!

**Pull request merged**